python_modules/dagster/dagster/_core/storage/event_log/base.py

```diff
@@ -634,3 +635,12 @@ def handles_run_events_in_store_event(self) -> bool:
 
     def default_run_scoped_event_tailer_offset(self) -> int:
         return 0
+
+    @abstractmethod
+    def get_updated_asset_status_cache_values(
+        self,
+        asset_keys: Sequence[AssetKey],
+        partitions_def: Sequence[Optional[PartitionsDefinition]],
+    ) -> Mapping[AssetKey, Optional[AssetStatusCacheValue]]:
```

Comment:
just use Optional["AssetStatusCacheValue"] here if we're not actually using this class for anything other than typing -- helps avoid circular import issues in the future